lib/em-http/http_connection.rb

```diff
@@ -40,36 +40,103 @@ def unbind(reason=nil)
       # https://github.com/lostisland/faraday/blob/63cf47c95b573539f047c729bd9ad67560bc83ff/lib/faraday/adapter/em_http_ssl_patch.rb
+      #
+      # Updated by Ably, here’s why:
+      #
+      # We noticed that the existing verification mechanism is failing in the
+      # case where the certificate chain presented by the server contains a
+      # certificate that’s signed by an expired trust anchor. At the time of
+      # writing, this is the case with some Let’s Encrypt certificate chains,
+      # which contain a cross-sign by the expired DST Root X3 CA.
+      #
+      # This isn’t meant to be an issue; the certificate chain presented by the
+      # server still contains a certificate that’s a trust anchor in most
+      # modern systems. So in the case where this trust anchor exists, OpenSSL
+      # would instead construct a certification path that goes straight to that
+      # anchor, bypassing the expired certificate.
+      #
+      # Unfortunately, as described in
+      # https://github.com/eventmachine/eventmachine/issues/954#issue-1014842247,
+      # EventMachine misuses OpenSSL in a variety of ways. One of them is that
+      # it does not configure a list of trust anchors, meaning that OpenSSL is
+      # unable to construct the correct certification path in the manner
+      # described above.
+      #
+      # This means that we end up in a degenerate situation where
+      # ssl_verify_peer just receives the certificates in the chain provided by
+      # the peer. In the scenario described above, one of these certificates is
+      # expired and hence the existing verification mechanism, which "verifies"
+      # each certificate provided to ssl_verify_peer, fails.
+      #
+      # So, instead we remove the existing ad-hoc mechanism for verification
+      # (which did things I’m not sure it should have done, like putting
+      # non-trust-anchors into an OpenSSL::X509::Store) and instead employ
+      # OpenSSL (configured to use the system trust store, and hence able to
+      # construct the correct certification path) to do all the hard work of
+      # constructing the certification path and then verifying the peer
+      # certificate. (This is what, in my opinion, EventMachine ideally would
+      # be allowing OpenSSL to do in the first place. Instead, as far as I can
+      # tell, it pushes all of this responsibility onto its users, and then
+      # provides them with an insufficient API for meeting this
+      # responsibility.)
       def ssl_verify_peer(cert_string)
-        cert = nil
+        # We use ssl_verify_peer simply as a mechanism for gathering the
+        # certificate chain presented by the peer. In ssl_handshake_completed,
+        # we’ll make use of this information in order to verify the peer.
+        @peer_certificate_chain ||= []
         begin
           cert = OpenSSL::X509::Certificate.new(cert_string)
+          @peer_certificate_chain << cert
+          true
```

Comment:
Do we need to delete this line 🤔
``` 
true
```

Reply:
Nope, the last expression executed in a method in Ruby is its return value. It's quite idiomatic to not write `return` (you’ll see this idiom used elsewhere in this file). I don't particularly like this feature of the language, but I thought best to be consistent with what's already there.

Reply:
I am totally new to ruby syntax, seems there's lot of syntactical sugar here

Reply:
Some people love Ruby; I’m not so much of a fan 😬